static/app/views/dashboardsV2/widgetBuilder/widgetBuilder.tsx

```diff
@@ -711,15 +717,15 @@ function WidgetBuilder({
 
           newQuery.orderby = `${orderbyPrefix}${newOrderByValue}`;
         } else {
-          const isFromAggregates = fieldStrings.includes(rawOrderby);
+          const isFromAggregates = newQuery.aggregates.includes(rawOrderby);
           const isCustomEquation = isEquation(rawOrderby);
           const isUsedInGrouping = newQuery.columns.includes(rawOrderby);
 
           const keepCurrentOrderby =
             isFromAggregates || isCustomEquation || isUsedInGrouping;
-          const firstAggregateAlias = isEquation(fieldStrings[0])
-            ? `equation[${getNumEquations(fieldStrings) - 1}]`
-            : fieldStrings[0];
+          const firstAggregateAlias = isEquation(newQuery.aggregates[0])
```

Comment:
Thanks for catching this Ed. I added a default `?? ''` here so it doesn't error out. I noticed some sorting functionality that I wanted to fix but I'm going to leave it out of this PR and follow up soon since this is the smallest change I can make, maintaining the same behaviour, and avoid trying to fit other fixes into here.

Reply:
👍  sounds good to me!